### DISCUSSION TITLE
i want to consult a detail about the " func (cs *State) addVote" in consensus module

### DISCUSSION THREAD

**Discussion by gnehcein** [tendermint/tendermint#6822] [Q&A]
![addvote](https://user-images.githubusercontent.com/65959195/129320668-8b942239-f715-44cb-a9e0-06fbdf81ef4e.png)
In this case, if the blockID corresponds to the lockedBlock of the node, can I enter precommit directly? The enterPrecommit also checks to see if blockID corresponds to lockedBlock.
So I think we can add a judgment, if that's the case, enterPrecommit, not enterPrevoteWait.
Sorry I am a beginner, if there is anything wrong please correct.

**Comment by williambanfield** [comment]
Hey @gnehcein, thanks for the question. 

In the case you're examining, a validator has seen >2/3 of the voting power on the network. If we end up in the case where we `enterPrevoteWait` on line `2162`, we have seen >2/3 of the voting power on the network but that >2/3 is not for a particular block. 

This validator has also not hit the round timeout. The reason we go to `enterPrevoteWait` even if we have a locked block is that we haven't hit the round timeout and nor have we seen >2/3 prevotes for a block or >2/3 prevotes for nil. Additional votes may bring us to either of those conditions and change what we should vote for in the round. We will not see a 2/3 majority in the round and precommit nil instead of for a block [here](https://github.com/tendermint/tendermint/blob/267aac2e9095cbae113ed6ee55e34d4bf8367ae8/internal/consensus/state.go#L1391). If we don't wait we effectively ignore the votes of slow but correct validators.

**Reply by gnehcein** [reply]
hello friend,what I mean by that is whether we can set a judgment in the case of 2/3major: If BlockID is not nil and corresponds to lockedBlock, we go straight to EnterPreCommit
